include/libdnf/common/exception.hpp

```diff
@@ -160,6 +160,24 @@ class SystemError : public Error {
 };
 
 
+/// Configuration file either is missing, we don't have read permission, or is otherwise inaccessible
+class InaccessibleConfigError : public Error {
+public:
+    using Error::Error;
+    const char * get_domain_name() const noexcept override { return "libdnf::base"; }
+    const char * get_name() const noexcept override { return "InaccessibleConfigError"; }
+};
+
+
+/// Couldn't parse configuration file
+class InvalidConfigError : public Error {
+public:
+    using Error::Error;
+    const char * get_domain_name() const noexcept override { return "libdnf::base"; }
+    const char * get_name() const noexcept override { return "InvalidConfigError"; }
+};
```

Comment:
It seems to me that these specific errors do not fit in this header file where more general errors are defined.
`include/libdnf/conf/config_parser.hpp` might be a better fit for them, what do you think?
After all, config parser is the place where these exceptions has their roots. Maybe, they can be even handled / thrown directly in the `ConfigParser::read()`?

Reply:
In case of adding the exceptions into some header which is already included in SWIG, like the `include/libdnf/conf/config_parser.hpp`, make sure to ignore these exceptions in related `*.i` file, otherwise it won't compile, e.g. in `conf.i` the following should do the trick:

```
%ignore libdnf::InaccessibleConfigError;
%ignore libdnf::InvalidConfigError;
```